c/public/lib/source/ttread/ttread.c

```diff
@@ -820,21 +993,551 @@ static void hmtxRead(ttrCtx h) {
+            result = 0; break;
+        }
+        if (peakTuple[i] == h->vf.ndv[i]) /* means no change in scalar value. */
+            continue;
+        if (tupleIndex & gvar_FLAG_INTERMEDIATE_TUPLE) { /* check if its an intermediate tuple. */
+            if (h->vf.ndv[i] < imStart[i] ||  h->vf.ndv[i] > imEnd[i]) {
+                result = 0; break;
+            } else if (imStart[i] > peakTuple[i] || peakTuple[i] > imEnd[i])
+                continue;
+            else if (imStart[i] < 0 && imEnd[i] > 0 && peakTuple[i] != 0)
+                continue;
+            else if (h->vf.ndv[i] < peakTuple[i]) {
+                if (peakTuple[i] != imStart[i])
+                    result = fixmul(result, fixdiv((h->vf.ndv[i] - imStart[i]), (peakTuple[i] - imStart[i])));
+            } else if (peakTuple[i] != imEnd[i])
+                result = fixmul(result, fixdiv((imEnd[i] - h->vf.ndv[i]), (imEnd[i] - peakTuple[i])));
+        } else { /* not an intermediate tuple. */
+            /* if selected instance is out of range on any axis, then the overall scalar for that delta will be 0. */
+            if (h->vf.ndv[i] < MIN(0, peakTuple[i]) || h->vf.ndv[i] > MAX(0, peakTuple[i])) {
+                result = 0; break;
+            }
+            result = fixmul(result, fixdiv(h->vf.ndv[i], peakTuple[i])); /* peakTuple[i] cannot be zero here. */
+        }
+    }
+
+    return result;
+}
+
+static unsigned long gvarReadPackedDeltas(ttrCtx h, int16_t* deltas, long deltaCount) {
+    unsigned int index = 0;
+    uint16_t controlByte;
+    uint16_t runCount;
+    unsigned int j;
+
+    while ( index < deltaCount ) {
+        controlByte = read1(h);
+        runCount = (controlByte & gvar_DELTA_RUNCOUNT_MASK);
+
+        if (controlByte & gvar_DELTAS_ARE_ZERO) {
+            /* 'runCount + 1' deltas are zero. */
+            for (j = 0; j <= runCount && index < deltaCount; j++)
+            deltas[index++] = 0;
+        } else if (controlByte & gvar_DELTAS_ARE_WORDS) {
+            /* read 'runCount + 1' deltas as two bytes. */
+            for (j = 0; j <= runCount && index < deltaCount; j++)
+                deltas[index++] = sread2(h);
+        } else {
+            /* read 'runCount + 1' deltas as one bytes. */
+            for (j = 0; j <= runCount && index < deltaCount; j++)
+                deltas[index++] = (int16_t)sread1(h);
+        }
+
+        if (j <= runCount)
+            fatal(h, ttrErrBadGlyphData, "invalid delta run count");
+    }
+
+    return deltaCount;
+}
+
+static void gvarInterpolateIntermCoord(
+    int untouch1, int untouch2,
+    int touch1, int touch2,
+    int16_t* coords,
+    Fixed *deltas) {
+    int p;
+    int v;
+    Fixed delta, delta1, delta2;
+    Fixed scale = 0;
+
+    if (untouch1 > untouch2)
+        return;
+
+    if (coords[touch1] > coords[touch2]) {
+        p = touch1;
+        touch1 = touch2;
+        touch2 = p;
+    }
+
+    delta1 = deltas[touch1];
+    delta2 = deltas[touch2];
+    delta = 0;
```

Comment:
This value stored in `delta` is overwritten before it is ever used, so this line can be removed.

Reply:
removed, and also minimized scope of variable `delta`